src/intensity_bias_filter.cpp

```diff
@@ -0,0 +1,77 @@
+#include <algorithm>
+#include "laser_filters/intensity_bias_filter.h"
+#include <ros/ros.h>
+
+laser_filters::IntensityBiasFilter::IntensityBiasFilter() {
+
+}
+
+bool laser_filters::IntensityBiasFilter::configure() {
+  bool num_total_beams_set = getParam("num_total_beams", num_total_beams_);
+  bool high_intensity_thresh_set = getParam("intensity_threshold", high_intensity_threshold_);
+  bool max_num_high_intensity_beams_set = getParam("max_num_high_intensity_beams", max_num_high_intensity_beams_);
+
+  if (max_num_high_intensity_beams_ > num_total_beams_) {
+      ROS_WARN_STREAM("Maximum beams chosen by intensity (" << max_num_high_intensity_beams_ <<") is greater than total beams(" << num_total_beams_ <<")");
+  }
+
+  return num_total_beams_set && high_intensity_thresh_set && max_num_high_intensity_beams_set;
+}
+
+bool laser_filters::IntensityBiasFilter::update(
+    const sensor_msgs::LaserScan& scan_in, sensor_msgs::LaserScan& scan_out)
+{
+    // all of the laser filters do this to save processing?
+    scan_out = scan_in;
+
+    // calculate how many high intensity beams to select
+    int total_num_high_intensity_beams = std::count_if(scan_out.intensities.begin(), scan_out.intensities.end(), [&](int i){return i > high_intensity_threshold_; });
```

Comment:
Conversion from float to int is undefined if the number is Nan (or any other exceptional value). Changing the lambda parameter type to float would be better (assuming a LaserScan can contain NaN)

Reply:
Done.